### ISSUE TITLE
ui: charts don't plot but tooltip shows correct data

### ISSUE THREAD

**Issue by vivace-io** [cockroachdb/cockroach#17936]
BUG REPORT

### Versions

Using official Docker Hub images, v1.0.4 and v1.0.5, OS/Arch Linux amd64 (Community). Docker version 17.06.0-ce build 02c1d87.

### What I did

Initially imported a production database from a v1.0.4 cluster via `cockroach dump` and imported the dump. Nothing special on that end.

I then ran a manual rolling update across a 4-node cluster, one node at a time by killing the container and starting a new one to take its place. After node 2 came back up and started to sync, I noted that the admin panel "Overview" section stopped updating several graphs (which I'll list below).

### Expected behavior

All graphs to render correctly and correspond with data displayed when hovering the mouse over the area in question.

### Actual behavior

The graphs themselves did not appear to update, however hovering the mouse across each empty section shows data that is present, just not being graphed. The data occasionally seems to be inaccurate. For example, one node may show 60+ replicas in for multiple ticks, and zero the next. In another instance, a node which was registered as dead and was recorded as having zero replicas eventually showed them present consistently for about 3 minutes before finally returning to zero.

I attempted to restart the cluster from a full stop, but nothing changed. I went through the logs of each node and as expected the only errors I saw were failures to locate the old nodes. I repeated the process as described above and was able to see the same issue, except on node 3's upgrade. I also ran the application against a blank database for v1.0.4 during the upgrade and under load, and experienced the same problem. No errors returned to the application (excluding "retry txn" obviously) and everything looks to be well-behaved, so I believe this is a UI issue but for the life of me I can't track it down.

### Missing graphs

- Service Latency (Cluster > Overview)
- Replicas Per Node (Cluster > Overview)
- Replicas per Store (Cluster > Replication)
- Leaseholders per Store (Cluster > Replication)
- Range Operations (Cluster > Replication) -- corrected itself.
- Snapshots (Cluster > Replication) -- corrected itself.
- Log Commit Latency (Cluster > Storage) -- later restored itself.
-- Command Commit Latency (Cluster > Storage) -- corrected itself.
- KV Transaction Duration (99th and 90th) (Cluster > Distrubted) -- corrected itself.

All other graphs not mentioned were fine. Graphs the corrected did so after roughly 1 hour.

Sorry if this was way too much detail, but it's been a slow day. Let me know if you need further details.

**Comment by tschottdorf** [comment]
Hi @vivace-io, thanks for your report. There are a number of things that can cause this. To narrow it down,

1. does the admin ui indicate that all nodes are healthy?
1. if so, make sure you can access the `yourhost:8080/debug` endpoint (may require the sql command `SET CLUSTER SETTING server.remote_debugging.mode = 'any';` (you may want to set this back to `off` when you're done)
1. see if you can access http://host:8080/#/reports/problemranges. If you can and there's anything there, send a copy of that page (you can save the HTML) and of all the "range info" pages linked to from it.
1. also attach the output of `./cockroach debug zip`.

Thank you!

**Comment by vivace-io** [comment]
All nodes are healthy, save for the dead ones for the rolling update. I did checkout `/debug` and saw nothing suspect (if there's something that could be of assistance I can paste here, let me know).  As for `/reports/problemranges` it returned status HTTP 200, with a blank body. Just the react element with nothing in it.

I'll try to replicate it once more and upload the debug tomorrow.

As a side note, I checked back after letting it sit for a while with a simulated load. The graphs are still not showing, but hovering the cursor over the graph boxes seems to display the correct metrics now for each node.

**Comment by tschottdorf** [comment]
This is an unsubstantiated idea, but do your Docker volumes not have persisted disk? I am assuming not because you say that in the UI you see three healthy nodes (corresponding to your running containers), but that you also have dead nodes (corresponding to the old containers) around. If you were really running without persistent storage, a rolling restart is almost certain to put your cluster at jeopardy since it amounts to killing replicas and replacing them by new nodes. If this is what happened, my assumption is that you have lost the majority of replicas for some of the replica sets housing time series data (and correspondingly this data is missing in the ui).

If you do want stateless containers, you'll need to use the decommissioning feature, which will be released in CockroachDB v1.1. It allows you to mark nodes for removal from the cluster, and to wait until it is safe to do so.

If you are using persistent storage, I'm curious why you would see any dead nodes in the ui (after all, you restarted them all).

**Comment by vivace-io** [comment]
I did use persistent disks for all nodes, so I'm also confused why they still show up.

I have left the cluster in question run since the initial update, and the graphs still are not plotting, but the data is 100% accurate when hovering the mouse over the UI as far as I can tell. Replicas are balanced as expected, latency for SQL requests are within my expected range, and so forth so I feel like it's definitely a UI issue. I also did point end up pointing Prometheus at the nodes and the data scraped seemed accurate.

I don't recall seeing a similar issue when upgrading from v1.0 to v1.0.4, but my memory is not what it should be.
